subsys/bluetooth/mesh/shell.c

```diff
@@ -911,12 +973,19 @@ static int cmd_ttl(const struct shell *shell, size_t argc, char *argv[])
 static int cmd_friend(const struct shell *shell, size_t argc, char *argv[])
 {
 	uint8_t frnd;
-	int err;
+	int err = 0;
 
 	if (argc < 2) {
 		err = bt_mesh_cfg_friend_get(net.net_idx, net.dst, &frnd);
 	} else {
-		uint8_t val = str2u8(argv[1]);
+		uint8_t val = isdigit(argv[1][0]) ?
+				      bt_mesh_shell_strtoul(argv[1], 0, &err) :
+					    (uint8_t)bt_mesh_shell_strtobool(argv[1], 0, &err);
```

Comment:
Why not just `bt_mesh_shell_strtobool`?

Reply:
I decided to try to not alter the functionality of the existing implementation. Since `friend_set` technically has three valid friend states (`disabled`, `enabled`, `not supported`) strtobool won't fully cover these possibillities like the previous one. I think this was the reason why the previous str2u8 was implemented in a peculiar manner. IMO the `not supported` state is pretty useless for the set call, so I agree that it would be sufficient to use strtobool anyways.

Reply:
The same goes for `relay set` and `gatt proxy set` aswell

Reply:
I guess `not supported` can't be used in this case. It is a configuration message which can either enable or disable a feature.

Reply:
Yes, I changed it for the three mentioned instances. I belive it was originally implemented like this because the spec does not clearly define which states are intended for set calls and which is only used in status responses.